src/ucp/api/ucp.h

```diff
@@ -1956,6 +1956,36 @@ ucs_status_t ucp_worker_fence(ucp_worker_h worker);
  */
 ucs_status_t ucp_worker_flush(ucp_worker_h worker);
 
+typedef enum {
+    UCP_ATOMIC_POST_OP_ADD,
+    UCP_ATOMIC_POST_OP_XOR,
+    UCP_ATOMIC_POST_OP_AND,
+    UCP_ATOMIC_POST_OP_OR,
+    UCP_ATOMIC_POST_OP_LAST,
+} ucp_atomic_post_op_t;
+
+typedef enum {
+    UCP_ATOMIC_FETCH_OP_FADD,
+    UCP_ATOMIC_FETCH_OP_SWAP, /* swap with *result */
+    UCP_ATOMIC_FETCH_OP_CSWAP,
+    UCP_ATOMIC_FETCH_OP_LAST,
+} ucp_atomic_fetch_op_t;
+
+ucs_status_t ucp_atomic_post32(ucp_ep_h ep, ucp_atomic_post_op_t opcode, uint32_t value,
+                               uint64_t remote_addr, ucp_rkey_h rkey);
+
+ucs_status_ptr_t ucp_atomic_fetch32_nb(ucp_ep_h ep, ucp_atomic_fetch_op_t opcode,
+                                       uint32_t value, uint32_t *result,
+                                       uint64_t remote_addr, ucp_rkey_h rkey,
+                                       ucp_send_callback_t cb);
+
+ucs_status_t ucp_atomic_post64(ucp_ep_h ep, ucp_atomic_post_op_t opcode, uint64_t value,
```

Comment:
@yosefe - maybe we should merge 32/64 AMO and add the branch. Maybe we can even inline it.